src/components/inject.js

```diff
@@ -317,7 +321,11 @@ Please be sure the component that calls createSource or createToken is within an
           return stripe.handleCardPayment(clientSecret, element);
         }
       } else {
-        if (!data || typeof data !== 'object') {
+        if (!data) {
+          throw new Error(
+            `Could not find a CardElement or CardNumberElement which which to perform handleCardPayment.`
+          );
+        } else if (typeof data !== 'object') {
```

Comment:
Are we sure object is strictly required? I thought you could pass a `clientSecret` and the id of a payment method or source as just a string?

Reply:
That does not seem to work, and the logic inside Stripe.js does not appear to support that.